api/src/role/role.service.ts

```diff
@@ -0,0 +1,187 @@
+        statusCode: HttpStatus.OK,
+        data: { roles },
+      };
+    } catch (error) {
+      if (error instanceof HttpException) {
+        return {
+          message: error.message,
+          statusCode: HttpStatus.CONFLICT,
+          data: {},
+        };
+      } else {
+        return {
+          message: error.message,
+          statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
+          data: {},
+        };
+      }
+    }
+  }
+
+  async update(
+    roleDto: RoleDto,
+    user: { userId: any },
+    id: number,
+  ): Promise<{ message: string; statusCode: number; data: object }> {
+    const { name } = roleDto;
+    try {
+      const existingRole = await this.roleModel.findOne({
+        where: { id: id },
+      });
+      if (existingRole) {
+        const existingUserRole = await this.roleModel.findOne({
+          where: { name: name },
+        });
+        if (
+          existingUserRole &&
+          existingUserRole.id.toString() !== id.toString()
+        ) {
+          throw new UnprocessableEntityException('Name already exists.');
+        }
+        existingRole.name = name;
+        existingRole.updatedBy = user.userId;
+        await existingRole.save();
+
+        const roles = await this.roleModel.findAll({
+          where: { status: { [Op.or]: [1, 2] } },
+        });
+
+        return {
+          message: 'Role updated successfully.',
+          statusCode: HttpStatus.OK,
+          data: roles,
+        };
+      } else {
+        return {
+          message: 'Role not found.',
+          statusCode: HttpStatus.NOT_FOUND,
+          data: {},
+        };
+      }
+    } catch (error) {
+      if (error instanceof HttpException) {
+        return {
+          message: error.message,
+          statusCode: HttpStatus.CONFLICT,
+          data: {},
+        };
+      } else {
+        return {
+          message: error.message,
+          statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
+          data: {},
+        };
+      }
+    }
+  }
+
+  async delete(
+    id: number,
+  ): Promise<{ message: string; statusCode: number; data: object }> {
+    try {
+      const role = await this.roleModel.findOne({
+        where: {
+          id,
+          status: true,
+        },
+      });
+
+      if (role) {
+        role.status = false;
+        await role.save();
+
+        const roles = await this.roleModel.findAll({
+          where: { status: true },
+        });
+
+        return {
+          message: 'Role deleted successfully.',
+          statusCode: HttpStatus.OK,
+          data: roles,
+        };
+      } else {
+        return {
+          message: 'Role not found.',
+          statusCode: HttpStatus.NOT_FOUND,
+          data: {},
+        };
+      }
+    } catch (error) {
+      return {
+        message: 'An error occurred while deleting the role.',
+        statusCode: HttpStatus.INTERNAL_SERVER_ERROR,
+        data: {},
+      };
+    }
+  }
+
+  async getActiveRoles(): Promise<Role[]> {
```

Comment:
Check the function and remove if not needed

Reply:
function is needed